rasterio/vrt.py

```diff
@@ -244,4 +244,58 @@ def _boundless_vrt_doc(
         dstrect.attrib['xSize'] = str(src_dataset.width)
         dstrect.attrib['ySize'] = str(src_dataset.height)
 
-    return ET.tostring(vrtdataset).decode('ascii')
+    return ET.tostring(vrtdataset, encoding="utf-8")
+
+
+class VirtualDataset:
+    """A lazily evaluated virtual dataset based on GDAL's VRT."""
+
+    @classmethod
+    def fromstring(cls, text):
+        """Create a virtual dataset from XML.
+
+        Paramters
+        ---------
+        text : bytes or str
+            XML as a byte or unicode string.
+
+        Returns
+        -------
+        VirtualDataset
+
+        """
+        # To avoid a circular import.
+        from rasterio.io import MemoryFile
+
+        tree = ET.fromstring(text)
+        doc = ET.tostring(tree)
+        memfile = MemoryFile(doc, ext=".vrt")
+        vrt = cls()
+        vrt._tree = tree
```

Comment:
Are `_tree` and `_doc` leftovers from a different idea?

Reply:
@snowman2 I'm thinking that exposing one or the other as an attribute gives us the most power to modify the VRT. Probably the infoset tree because it's mutable.